eclair-core/src/main/scala/fr/acinq/eclair/payment/receive/MultiPartHandler.scala

```diff
@@ -48,6 +48,52 @@ class MultiPartHandler(nodeParams: NodeParams, db: IncomingPaymentsDb, commandBu
    */
   def doHandle(paymentHash: ByteVector32): Boolean = true
 
+  /**
+   * Fulfill all parts of an incoming payment.
+   * Can be overridden for a more fine-grained control of when and how to fulfill.
+   */
+  def doFulfill(ctx: ActorContext, preimage: ByteVector32, e: MultiPartPaymentFSM.MultiPartPaymentSucceeded)(implicit log: LoggingAdapter): Unit = {
```

Comment:
The current changes are enough for the hodl plugin invoices and likely for most use cases of the incoming payments. I highlighted here a change that could simplify even more the life of the plugin developers:
```suggestion
  def doFulfill(system: ActorSystem, preimage: ByteVector32, e: MultiPartPaymentFSM.MultiPartPaymentSucceeded)(implicit log: LoggingAdapter): Unit = {
```
This change lets the plugin developer avoid to reimplement `doFulfill` entirely, instead it can use `super.doFulfill` when it wants to fulfill the HTLC. That's necessary because a subclass of `MultiPartHandler` is not an actor and doesn't have an actor context but it does have the actor system which is enough to access the event stream. I'm not strongly in favor of this change but is be a nice-to-have.

Reply:
Done, it was an easy low hanging fruit.